### PR TITLE
Fix container name problem and rename struct to avoid ambiguousness

### PR DESCRIPTION
**Pull request by stanlyxiang**
1.replace magic number
2.check container name instead of confused container name rule
3.rename plcContainer -> plcContainerConf to avoid ambiguousness

### CODE REVIEW

File: src/containers.c [hunk 1]
```diff
@@ -345,5 +351,24 @@ char *parse_container_meta(const char *source) {
...
     name[nameptr] = '\0';
 
+    int regt = check_container_name(name);
+    if (regt == -1) {
+        lprintf(ERROR, "Container name '%s' contains illegal character for container.", name);
+    }
+
     return name;
 }
+
```
Comment (HaozhouWang) [inline src/containers.c]: It could be better to add some comments about this function

File: src/containers.c [hunk 2]
```diff
@@ -345,5 +351,24 @@ char *parse_container_meta(const char *source) {
     }
     name[nameptr] = '\0';
 
+    int regt = check_container_name(name);
+    if (regt == -1) {
```
Comment (HaozhouWang) [inline src/containers.c]: maybe using boolean instead of -1

File: src/containers.c [hunk 3]
```diff
@@ -320,20 +321,25 @@ char *parse_container_meta(const char *source) {
...
-    name = (char*)pmalloc(last-first+1);
-    while (first < last && source[first] != ':') {
-        if (!is_whitespace(source[first])) {
-            name[nameptr] = source[first];
-            nameptr++;
-        }
+    name = (char*)pmalloc(last-first+1 + 1);
+    while (first <= last ){
+        name[nameptr] = source[first];
+        nameptr++;
```
Comment (paul-guo-) [inline src/containers.c]: Use memcpy instead.

File: src/containers.c [hunk 4]
```diff
@@ -320,20 +321,25 @@ char *parse_container_meta(const char *source) {
...
+     * the character length of name is last-first+1
+     * +1 for terminator
      */
-    name = (char*)pmalloc(last-first+1);
-    while (first < last && source[first] != ':') {
-        if (!is_whitespace(source[first])) {
-            name[nameptr] = source[first];
-            nameptr++;
-        }
+    name = (char*)pmalloc(last-first+1 + 1);
```
Comment (paul-guo-) [inline src/containers.c]: last - first + 1 + 1

File: src/containers.c [hunk 5]
```diff
@@ -345,5 +351,24 @@ char *parse_container_meta(const char *source) {
...
+    regex_t    re;
+    if (regcomp(&re, "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$", REG_EXTENDED | REG_NOSUB | REG_NEWLINE) != 0) {
+        return -1;
+    }
+    status = regexec(&re, name, (size_t) 0, NULL, 0);
+    regfree(&re);
+    if (status != 0) {
+        return -1;
+    }
+    return 1;
```
Comment (paul-guo-) [inline src/containers.c]: I'd it returns 0 for OK case which is the convention of UNIX.

File: src/plcontainer.c [hunk 6]
```diff
@@ -170,7 +170,7 @@ static plcProcResult *plcontainer_get_result(FunctionCallInfo  fcinfo,
     name = parse_container_meta(req->proc.src);
     conn = find_container(name);
     if (conn == NULL) {
-        plcContainer *cont = NULL;
+        plcContainerConf *cont = NULL;
```
Comment (paul-guo-) [inline src/plcontainer.c]: Better cont -> conf for all?

File: src/containers.c (hunk 2 above)
Comment (stanlyxiang) [inline src/containers.c]: just follow other functions.  -1 for failure and 0 for success.

**Comment by paul-guo-** [comment]
+1

**Comment by davecramer** [comment]
What issue is this addressing ?

**Comment by stanlyxiang** [comment]
@davecramer 
before this fix: 
I can define container name in function like this:   #container: con  tainer _  name.  Plcontainer will find container definition in xml file use name "container_name"( remove blank spaces and combine them together. However I must define "container_name" (without spaces) in xml files. This mismatch is  too confused.  
after this fix: container name in function definition and xml file should be same.  The name rule follow the docker container name convention. 

Do you have idea about the name convention ? Is it too strict ?

**Comment by stanlyxiang** [comment]
@davecramer 
Thank you Dave. We will explain what PRs do in the PRs so that everyone can know the problems and fixes.

**Pull request merged**